### PR TITLE
On X11, replay modifiers consumed by XIM

### PR DESCRIPTION
**Pull request by kchibisov**
- [x] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by kchibisov** [comment]
There're multiple approach to that, we can (in theory) check if the event is a modifier and handle it before `XIM` and then just discard it once it's arrives from XIM. But this can probably desync us, so the solution is just emit a bit more, though it shouldn't really matter, it's not like we can do better here.

Modifiers will basically go like `Shift(xkb) -> empty(core, replay) -> key(xim forwarded) -> Shift(restoring)`. Not doing so will result in e.g. `Shift + shift` bindings not working, since you can't detect such thing at all.

File: src/platform_impl/linux/x11/event_processor.rs [hunk 1]
```diff
@@ -41,6 +41,9 @@ use crate::platform_impl::x11::{
 /// The X11 documentation states: "Keycodes lie in the inclusive range `[8, 255]`".
 const KEYCODE_OFFSET: u8 = 8;
 
+/// The maximum amount of X modifiers to replay.
+const MAX_MOD_REPLAY_LEN: usize = 32;
```
Comment (notgull) [inline src/platform_impl/linux/x11/event_processor.rs:45]:
```suggestion
pub(super) const MAX_MOD_REPLAY_LEN: usize = 32;
```

File: src/platform_impl/linux/x11/mod.rs [hunk 2]
```diff
@@ -322,6 +325,8 @@ impl<T: 'static> EventLoop<T> {
             ime_receiver,
             ime_event_receiver,
             xi2ext,
+            xfiltered_modifiers: Default::default(),
```
Comment (notgull) [inline src/platform_impl/linux/x11/mod.rs:328]:
```suggestion
            xfiltered_modifiers: std::collections::VecDeque::with_capacity(event_processor::MAX_MOD_REPLAY_LEN),
```

**Review by notgull — APPROVED** [review]
Minor nit

**Pull request merged**